File: 101-200/101-110/102-BSTLevelOrderTraversal/BSTLevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
         
        queue = []
        result = []
        queueFront = queueBack = 0
        
        if root:
            queue.append(root)
            queueFront += 1
            result.append([root.val])
        
        while queueBack < queueFront:
            increase = 0
            while queueBack < queueFront:
                node = queue[queueBack]
                queueBack += 1
                if node.left:
                    queue.append(node.left)
                    increase += 1
                if node.right:
                    queue.append(node.right)
                    increase += 1
            if increase:
                queueFront += increase
                result.append(map(lambda x: x.val, queue[queueBack: queueFront]))
        
        return result

File: 101-200/101-110/102-BSTLevelOrderTraversal/BSTLevelOrderTraversal.cpp
#include <queue>
#include <vector>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

using namespace std;

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        if (!root)
            return result;

        queue<TreeNode *> prev_level, next_level;
        prev_level.push(root);

        while (!prev_level.empty()) {
            vector<int> int_level;
            while (!prev_level.empty()) {
                TreeNode *node = prev_level.front();
                prev_level.pop();

                if (node){
                    int_level.push_back(node->val);
                    next_level.push(node->left);
                    next_level.push(node->right);
                }
            }

            result.push_back(int_level);
            next_level.swap(prev_level);
        }

        result.pop_back();
        return result;
    }
};